### PR TITLE
Refactor: move Commit To APIFormat Code & Lot of StopWatch related things

### LINKED ISSUE #12723: [Note] Move StopWatch.APIFormat() to convert package
from https://github.com/go-gitea/gitea/pull/12458#issuecomment-687498000
...

+ cache Issues and Repos if convert a list of StopWatches to APIFormat ...

### PR DESCRIPTION
**Pull request by 6543**
close #12723

## StopWatch
* extend TestAPIListStopWatches
* remove type Stopwatches

 ## Commit:
* create own file for related things in convert package
* rename ToCommit ToPayloadCommit
* move toCommit into convert package

### CODE REVIEW

**Review by zeripath — APPROVED** [review]

File: modules/convert/git_commit.go [hunk 1]
```diff
@@ -0,0 +1,165 @@
+// Copyright 2020 The Gogs Authors. All rights reserved.
```
Comment (lunny) [inline modules/convert/git_commit.go]: Gitea

**Review by lunny — CHANGES_REQUESTED** [review]
Otherwise LGTM

**Comment by lunny** [comment]
And I think this could be moved to v1.13 once it's ready.

File: modules/convert/git_commit.go (hunk 1 above)
Comment (6543) [inline modules/convert/git_commit.go]:
```suggestion
// Copyright 2020 The Gitea Authors. All rights reserved.
```

Comment (6543) [reply modules/convert/git_commit.go]: @lunny done

**Review by lunny — APPROVED** [review]

**Pull request merged**